src/test/java/com/conveyal/gtfs/loader/JDBCTableWriterTest.java

```diff
@@ -421,6 +422,60 @@ public void canCreateUpdateAndDeleteScheduleExceptions() throws IOException, SQL
         assertThatSqlQueryYieldsZeroRows(getColumnsForId(scheduleException.id, scheduleExceptionTable));
     }
 
+    /**
+     * Ensure that {@link ScheduleException}s which are loaded from an existing GTFS can be removed properly,
+     * including created entries in calendar_dates.
+     */
+    @Test
+    public void canCreateAndDeleteCalendarDates() throws IOException, SQLException, InvalidNamespaceException {
+        String firstServiceId = "REMOVED";
+        String secondServiceId = "ADDED";
+        String[] allServiceIds = new String[]{firstServiceId, secondServiceId};
+        String[] holidayDates = new String[]{"20190812", "20190813", "20190814"};
+
+        final Table scheduleExceptionTable = Table.SCHEDULE_EXCEPTIONS;
+        final Table calendarDatesTable = Table.CALENDAR_DATES;
+        final Class<ScheduleExceptionDTO> scheduleExceptionDTOClass = ScheduleExceptionDTO.class;
+
+        // Create new schedule exception which involves 2 service IDs and multiple dates
+        ScheduleExceptionDTO exceptionInput = new ScheduleExceptionDTO();
+        exceptionInput.name = "Incredible multi day holiday";
+        exceptionInput.exemplar = 9; // Add, swap, or remove type
+        exceptionInput.removed_service = new String[]{firstServiceId};
+        exceptionInput.added_service = new String[]{secondServiceId};
+        exceptionInput.dates = holidayDates;
+
+        // Save the schedule exception
+        // TODO: share this with other schedule exception method?
+        TableWriter<ScheduleException> createTableWriter = createTestTableWriter(scheduleExceptionTable);
+        String scheduleExceptionOutput = createTableWriter.create(mapper.writeValueAsString(exceptionInput), true);
+        ScheduleExceptionDTO scheduleException = mapper.readValue(scheduleExceptionOutput, scheduleExceptionDTOClass);
+
+        // Create a calendar_dates entry for each date of the schedule exception
+        for (String date: holidayDates) {
+            createAndStoreCalendarDate(firstServiceId, date, 2); // firstServiceId is removed
+            createAndStoreCalendarDate(secondServiceId, date, 1); // secondServiceId is added
+        }
+
+        // Delete a schedule exception
+        JdbcTableWriter deleteTableWriter = createTestTableWriter(scheduleExceptionTable);
+        int deleteOutput = deleteTableWriter.delete(scheduleException.id, true);
+        LOG.info("deleted {} records from {}", deleteOutput, scheduleExceptionTable.name);
+
+        // Verify that the entries in calendar_dates are removed after deleting the schedule exception.
+        for (String date : holidayDates) {
+            for (String serviceId : allServiceIds){
+                String sql = String.format("select * from %s.%s where service_id = '%s' and date = '%s'",
```

Comment:
Or you could use a prepared statement as you did in class `JdbcTableWriter`.

Reply:
I did it like this to mimic the other tests above, where a sql string is constructed and then passed to the `assertThatSqlQueryYieldsZeroRows` method. I can use a prepared statement if you think it improves this though.

Reply:
Ok I'll keep as is then.